File: gdscript/src/main/java/gdscript/GdFileType.java
package gdscript;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class GdFileType extends LanguageFileType {

    public static final GdFileType INSTANCE = new GdFileType();

    private GdFileType() {
        super(GdLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "GdScript File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "GdScript language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "gd";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return GdIcon.FILE;
    }

}

File: gdscript/src/main/kotlin/gdscript/GdFileType.kt
package gdscript

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object GdFileType : LanguageFileType(GdLanguage) {
    override fun getName(): String = GdLanguage.id
    override fun getDescription(): String = GdScriptBundle.message("language.file_name")
    override fun getDefaultExtension(): String = "gd"
    override fun getIcon(): Icon = GdScriptPluginIcons.Icons.GDScript
}
